docs/content/sips/xxx-manifest-redesign.md

```diff
@@ -0,0 +1,123 @@
+title = "SIP xxx - Application Manifest Redesign"
+template = "main"
+date = "2022-05-20T13:22:30Z"
+---
+
+Summary: A new design for Application Manifests (`spin.toml`)
+
+Owner: lann.martin@fermyon.com
+
+Created: May 20, 2022
+
+## Background
+
+Spin's Application Manifest file (usually named `spin.toml`) describes a Spin application's metadata,
+components, and trigger configurations. The current design supports a single trigger type and combines
+component and trigger configuration with an implicit one-to-one relationship:
+
+```toml
+name = "hello-world"
+trigger = { type = "http", base = "/hello" }
+
+[[component]]
+id = "world"
+# ...
+[component.trigger]
+route = "/world"
+```
+
+In order to support future features, we would like to enable more flexible trigger configurations:
+
+- Multiple trigger types in an application
+- Multiple triggers associated with a single component
+- One trigger associated with multiple components
+
+## Proposal
+
+### Global trigger config
+
+In order to allow for multiple trigger types, we remove the top-level `trigger.type` field.
+We also move all application metadata from the top-level config to a new `[application]`
+section, and all "global trigger config" to new `[application.trigger.<type>]` sections:
+
+```toml
+[application]
+name = "hello-world"
+[application.trigger.http]
+base = "/hello"
+```
+
+> In the short term we can continue to enforce the use of exactly one trigger type in an
+> application as a manifest validation step.
+
+### Trigger config
+
+In order to decouple triggers and components, we move triggers to new top-level
+`[[trigger.<type>]]` sections:
+
+```toml
+[[trigger.http]]
+route = "/world"
+component = "world"
+
+[[component]]
+id = "world"
+# ...
+```
+
+### Manifest version
+
+Since this proposal represents a backward-incompatible change to the manifest format, the manifest
+version changes. The existing `spin_version = "1"` field name leaves some room for misinterpretation
+as "compatible with Spin project version 1". As an optional part of this proposal, change that name:
+
+```toml
+spin_manifest_version = "2"
+```
```

Comment:
Per discussion with @fibonacci1729 we don't _have_ to do this. See "Don't update the manifest version" below.

Reply:
I'm in favor of renaming spin_version to spin_manifest_version but I think we should wait to officially freeze version 1 manifest changes until spin 1.0

I'm hesitant toward bumping the manifest version before our first major release.

Reply:
Agree on this from an aesthetic point of view, but without some sort of schema ID, a user who downloads a new Spin build that changes the schema is going to get cryptic parse errors working with existing manifests.  Can we make sure that we do at least enough to give a meaningful error like "this version of the manifest is no longer supported"?

Reply:
Says the man who just changed templates.